llvm/tools/sycl-post-link/SpecConstants.cpp

```diff
@@ -543,7 +548,8 @@ Instruction *emitSpecConstantComposite(Type *Ty, ArrayRef<Value *> Elements,
 Instruction *emitSpecConstantRecursiveImpl(Type *Ty, Instruction *InsertBefore,
                                            SmallVectorImpl<ID> &IDs,
                                            unsigned &Index,
-                                           Constant *DefaultValue) {
+                                           Constant *DefaultValue,
+                                           const Module &M) {
```

Comment:
Module should be available either through `InsertBefore` or through `DefaultValue`. there is probably no need to pass it around through the whole calls stack